### PR TITLE
Rewrite cucumber tests in declarative style; fixes#64

### PR DESCRIPTION
**Pull request by alexander-kh**
Corresponds to PT story #136690541

Review request please. Any remarks concerning style, language and spelling are very welcome.

Notes:
* .best_practices.md and .useful_articles.md we can exclude from the repository and add in our project library
* in notes.md there are questions for discussion and points for improvement

### CODE REVIEW

File: features/.best_practices.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+"Connextra" format
```
Comment (mattlindsey) [inline features/.best_practices.md]: Move this file to doc folder.

File: features/.notes.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
+Logging in as a provider
```
Comment (mattlindsey) [inline features/.notes.md]: Move this file to doc folder.

File: features/.useful_articles.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+Cucumber anti-patterns (part one)
```
Comment (mattlindsey) [inline features/.useful_articles.md]: Move this file to doc folder.  Maybe combine with .best_practices.md file.

Comment (alexander-kh) [inline features/.useful_articles.md]: Matt, thanks for reviewing.
I've combined .best_practices.md and .useful_articles.md into one file and moved it to doc folder, as well as .notes_and_tasks.md

**Comment by alexander-kh** [comment]
Thanks for the attentive review.
I think it's a good suggestion, and present perfect is more suitable, though past simple is grammatically correct. There is subtle distinction in semantics.
The action (signing in) happened in the past, but important at the moment.

**Comment by alexander-kh** [comment]
I've revised the steps and the corresponding definitions.
Take a look before merging, please.

**Pull request merged**